packages/experimental/MenuButton/src/MenuButton.tsx

```diff
@@ -17,65 +18,15 @@ export const MenuButton = compose<MenuButtonType>({
   slots: {
     root: React.Fragment,
     chevronIcon: SvgXml,
+    button: Button,
   },
-  slotProps: {
-    root: {},
-  },
-  render: (props: MenuButtonProps, useSlots: UseSlots<MenuButtonType>) => {
-    const { menuItems, content, icon, disabled, onItemClick, contextualMenu, primary, style } = props;
-
-    const stdBtnRef = useRef(null);
-    const [showContextualMenu, setShowContextualMenu] = useState(false);
-    const onDismiss = useCallback(() => {
-      setShowContextualMenu(false);
-    }, [setShowContextualMenu]);
-    const toggleShowContextualMenu = useCallback(() => {
-      setShowContextualMenu(!showContextualMenu);
-    }, [showContextualMenu, setShowContextualMenu]);
-
-    const buttonProps = {
-      content,
-      disabled,
-      primary,
-      icon,
-      style,
-      componentRef: stdBtnRef,
-      onClick: toggleShowContextualMenu,
-    };
-
-    const contextualMenuProps = {
-      onItemClick,
-      target: stdBtnRef,
-      onDismiss,
-      setShowMenu: toggleShowContextualMenu,
-      ...contextualMenu,
-    };
-
-    const menuItemsUpdated = menuItems?.map((item) => {
-      if (item.hasSubmenu) {
-        const [showSubmenu, setShowSubmenu] = useState(false);
-        const toggleShowSubmenu = useCallback(() => {
-          setShowSubmenu(!showSubmenu);
-        }, [showSubmenu, setShowSubmenu]);
-        const onDismissSubmenu = useCallback(() => {
-          setShowSubmenu(false);
-        }, [setShowSubmenu]);
-        const { onHoverIn = toggleShowSubmenu, submenuProps = {}, ...restItems } = item;
-        const { onDismiss = onDismissSubmenu, setShowMenu = toggleShowSubmenu, ...restSubmenuProps } = submenuProps;
-        const menuItemUpdated = {
-          ...restItems,
-          onHoverIn,
-          showSubmenu: item.showSubmenu ?? showSubmenu,
-          submenuProps: { ...restSubmenuProps, onDismiss, setShowMenu },
-        };
-        return menuItemUpdated;
-      }
-      return item;
-    });
+  render: (userProps: MenuButtonProps, useSlots: UseSlots<MenuButtonType>) => {
+    const MenuButton = useMenuButton(userProps);
 
-    const Slots = useSlots(props);
+    const Slots = useSlots(MenuButton.props, layer => MenuButton.state[layer] || MenuButton.props[layer]);
 
-    return () => {
+    return (final: MenuButtonProps) => {
+      const { primary, contextualMenu, menuItems, ...mergedProps } = mergeProps(MenuButton.props, final);
```

Comment:
Does this mean there is a potential for onClick to get overridden if someone passes their own onClick into the MenuButton? That would make it so that the menu doesn't get toggled anymore, if I understand the system correctly?

Reply:
I tested this behavior. It wasn't changed